velox/functions/lib/Re2Functions.cpp

```diff
@@ -1031,28 +1122,42 @@ PatternMetadata determinePatternKind(StringView pattern) {
   // At this point pattern contains exactly one fixed pattern.
   // Pattern contains no wildcard characters (is a fixed pattern).
   if (wildcardStart == -1) {
-    return PatternMetadata{PatternKind::kFixed, patternLength};
+    auto unescapedPattern = unescape(pattern, 0, patternLength, escapeChar);
+    return PatternMetadata{
+        PatternKind::kFixed,
+        (vector_size_t)unescapedPattern.size(),
```

Comment:
vector_size_t should be used for row numbers and vector sizes, but not for size of a string. Why is it used here?

Reply:
Currently `PatternMetadata::length` is of type `vector_size_t `, will change it to `size_t`.